native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java

```diff
@@ -293,22 +301,69 @@ private static String[] getPathStringArray(BArray pathArray) {
 
     private static String constructQueryString(BMap params) {
         List<String> queryParams = new ArrayList<>();
+        Map<String, String> annotationValues = getQueryNameMapping(params);
         BString[] keys = (BString[]) params.getKeys();
         if (keys.length == 0) {
             return "";
         }
         for (BString key : keys) {
             Object value = params.get(key);
+            String queryName = key.getValue();
+            queryName = annotationValues.getOrDefault(queryName, queryName);
             String valueString = value.toString();
             if (value instanceof BArray) {
                 valueString = valueString.substring(1, valueString.length() - 1);
                 valueString = valueString.replace(QUOTATION_MARK, EMPTY);
             }
-            queryParams.add(key.getValue() + EQUAL_SIGN + valueString);
+            queryParams.add(queryName + EQUAL_SIGN + valueString);
         }
         return String.join(AND_SIGN, queryParams);
     }
 
+    /**
+     * This util function extracts the query name with the query annotation.
+     *
+     * @param params - Parameter map
+     * @return Map of string with overridden query param names
+     */
+    private static Map<String, String> getQueryNameMapping(BMap params) {
+        Map<String, String> annotationValues = new HashMap<>();
+        if (!(params.getType() instanceof RecordType queryRecord)) {
+            return annotationValues;
```

Comment:
Shall we add a test case to cover this part of the code? I think we can try to pass a map for queries

Reply:
Removed this check since we only allow the query with the QueryParam type.